map/bookmark_helpers.cpp

```diff
@@ -370,46 +371,49 @@ std::string GetKMLorGPXPath(std::string const & filePath)
     if (kmlData == nullptr)
       return {};
 
-    fileSavePath = GenerateValidAndUniqueFilePathForKML(base::FileNameFromFullPath(filePath));
+    auto fileSavePath = GenerateValidAndUniqueFilePathForKML(base::FileNameFromFullPath(filePath));
     if (!SaveKmlFileByExt(*kmlData, fileSavePath))
       return {};
+    return {std::move(fileSavePath)};
   }
   else if (fileExt == kKmzExtension)
   {
-    // Extract KML file from KMZ archive and save to temp KML with unique name.
+    // Extract KML files from KMZ archive and save to temp KMLs with unique name.
+    std::vector<std::string> kmlFilePaths;
     try
     {
       ZipFileReader::FileList files;
       ZipFileReader::FilesList(filePath, files);
-      std::string kmlFileName;
-      std::string ext;
-      for (size_t i = 0; i < files.size(); ++i)
+      std::string indexFileName;
+      std::vector<std::string> kmlFileNames;
+      files.erase(std::remove_if(files.begin(), files.end(),
+          [](auto const & file){ return GetLowercaseFileExt(file.first) != kKmlExtension; }),
+          files.end());
+
+      for (const auto & file : files)
       {
-        ext = GetLowercaseFileExt(files[i].first);
-        if (ext == kKmlExtension)
-        {
-          kmlFileName = files[i].first;
-          break;
-        }
+        auto kmlFileInZip = file.first;
```

Comment:
```suggestion
        auto const & kmlFileInZip = file.first;
```
Avoid creating unnecessary copies/doing unnecessary memory allocations.